Feat: Unified Task Scheduling Interface for "Schedule" view

Droyder7 opened the pull request:
# Description

This PR introduces a **unified task scheduling interface** that replaces the basic textarea with a smarter task selector. Users can now **create new tasks or schedule existing ones** from a single input within the schedule view.

<img width="296" height="187" alt="image" src="https://github.com/user-attachments/assets/2d88c12e-139c-4190-9fc6-dd8e0e719d7d" />

<img width="287" height="258" alt="image" src="https://github.com/user-attachments/assets/a8ef7f5d-9869-4700-8d4b-07cbe992255e" />


---

### 🚀 Key Enhancements

#### 1. **Unified Task Input**

* Previously: Only supported new task creation via plain text.
* Now:

  * Type to create a new task (if no match found)
  * Search and select from existing tasks with autocomplete

#### 2. **Improved Autocomplete**

* Smart filtering with real-time search
* Visual indicators for create vs. select modes
* Compact, responsive UI optimized for mobile and desktop

#### 3. **Dual Scheduling Modes**

* Toggle between **Day Planning** and **Time Scheduling** (via `Ctrl+1`)

#### 4. **Better State Management**

* Angular Signals for reactive state

  * `selectedTask`, `newTaskTitle`, `isCreate`, `isForDayMode`

#### 5. **Keyboard Support**

* `Enter`: Confirm create/select
* `Escape`: Cancel
* `Arrow Keys`: Navigate suggestions
* `Ctrl+1`: Toggle scheduling modes

---

### 🧪 Tested Scenarios

* Create new task and schedule it
* Select existing task and schedule it
* Toggle modes with `Ctrl+1`
* Handle empty/duplicate inputs and network errors

---

### ✅ User Impact

* Streamlined workflow (no mode switching)
* Prevents task duplication
* Faster, more intuitive task scheduling
* Better accessibility and responsiveness

---

## Issues Resolved

This PR resolves & implement the alternate solution mentioned in the issue: #4738 

## Check List

- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

johannesjo commented:
Thank you very much! 

Could be a great addition! There are some styling issues and it does not seem to work always: 

https://github.com/user-attachments/assets/cbf0731f-106d-4eed-a2ea-6816bc17950a

Droyder7 commented:
> Thank you very much!
> 
> Could be a great addition! There are some styling issues and it does not seem to work always:
> 
>  unified-task-interface-pr-2025-07-11_18.45.39.mov

Thanks for the feedback! That's a good catch. This seems to be happening when using mouse for selecting task. 

Navigating through list of task & selecting the task works just fine when using keyboard, that's why I also didn't notice it. 

I'll look into it & try to find a fix, so that when selecting a task from the suggestion list it doesn't loose focus & fail to schedule the task.

Droyder7 commented:
Hi @johannesjo, I've addressed the issue and fixed the inconsistent behavior in the latest commits.

✅ Autocomplete behaves consistently across different inputs
✅ Added checks to handle edge cases that were previously causing issues

attached recording for reference:

https://github.com/user-attachments/assets/30d95ed5-2405-4709-8168-4510142a8ef9


Let me know if it looks good on your end or if anything else needs adjustment!

Droyder7 commented on src/app/features/schedule/create-task-placeholder/create-task-placeholder.component.ts:
```diff
@@ -47,82 +54,221 @@ export class CreateTaskPlaceholderComponent implements OnDestroy {
+  readonly selectTaskMinimal = viewChild('selectTaskMinimal', { read: ElementRef });
+
+  private _interactionState: 'Idle' | 'Editing' | 'Selecting' = 'Idle';
+  private _blurTimeout: number | undefined;
```
fixed

Droyder7 commented on src/app/features/schedule/create-task-placeholder/create-task-placeholder.component.ts:
```diff
@@ -47,82 +54,221 @@ export class CreateTaskPlaceholderComponent implements OnDestroy {
+
+  onAutocompleteClosed(): void {
+    // Add a small delay to ensure option selection is processed first
+    setTimeout(() => {
```
implemented _selectionTimeout in the same pattern as _blurTimeout

Droyder7 commented on src/app/features/schedule/create-task-placeholder/create-task-placeholder.component.ts:
```diff
@@ -47,82 +54,221 @@ export class CreateTaskPlaceholderComponent implements OnDestroy {
+        });
         const task = await this._taskService.getByIdOnce$(id).toPromise();
+        if (!task) {
+          throw new Error(`Could not resolve task with id ${id}`);
```
fixed as suggested

Droyder7 commented on src/app/features/schedule/create-task-placeholder/create-task-placeholder.component.ts:
```diff
@@ -47,82 +54,221 @@ export class CreateTaskPlaceholderComponent implements OnDestroy {
-    } else if (event.key === '1' && event.ctrlKey) {
-      this.isForDayMode.set(!this.isForDayMode());
+    } catch (error) {
+      devError(`Failed to create/schedule task: ${error}`);
```
invoked snackService on error for providing user feedback

Droyder7 commented on src/app/features/tasks/select-task/select-task-minimal/select-task-minimal.component.ts:
```diff
@@ -0,0 +1,215 @@
+  filteredTasks$!: Observable<Task[]>;
+
+  private _destroy$ = new Subject<void>();
+  private _onChange: (value: Task | string) => void = (value: Task | string): void => {};
```
fixed as suggested

Droyder7 commented:
Hi @johannesjo, just pushed some changes addressing the PR comments. Please check it out

johannesjo commented:
Hey @Droyder7 ! Sorry I missed your comment. I'll have a look at it!

johannesjo commented:
Thank you for the warm words! :) It's really a cool addition! I will try to make a new release this Friday that will include this change.

johannesjo commented:
You can send them to contact@super-productivity.com 
Maybe you can post a part of them here, so I can verify it is really you?
